### PR TITLE
New recipe for seqgen-1.3.4

### PR DESCRIPTION
**Pull request by BenjaminDoran**

### CODE REVIEW

**Comment by BenjaminDoran** [comment]
hmm, build_tarballs worked locally on macos in docker. and looking at the logs it seems the build script seems to have exited successfully. But they all error with 

```log
[18:26:16]          ^
[18:26:16] cc  -O3 -o seq-gen seq-gen.o global.o evolve.o model.o nucmodels.o aamodels.o eigen.o treefile.o progress.o gamma.o twister.o -lm
[18:26:16]  ---> mkdir ${bindir}
[18:26:16]  ---> [[ "${target}" == *-mingw32* ]]
[18:26:16]  ---> cp seq-gen ${bindir}/seq-gen
[18:26:16]  ---> source "$_script"
[18:26:16]  ---> source "$_script"
[18:26:16] Child Process exited, exit code 0
[ Info: Beginning audit of /agent/_work/1/s/S/seqgen/build/i686-linux-musl/XXMOW7YL/i686-linux-musl-libgfortran3-cxx03/destdir
[ Info: Checking bin/seq-gen with RPath list Any[]
[ Info: Ignored system libraries libc.musl-i386.so.1
[ Info: Checking license file
[ Info: Found license file(s): license
[ Info: /agent/_work/1/s/S/seqgen/build/i686-linux-musl/XXMOW7YL/i686-linux-musl-libgfortran3-cxx03/destdir/bin/seq-gen matches our search criteria of ["seq-gen"]
[ Info: Compressing files in /agent/_work/1/s/S/seqgen/build/i686-linux-musl/XXMOW7YL/i686-linux-musl-libgfortran3-cxx03/destdir/logs/SeqGen
[ Info: Tree hash of contents of SeqGen.v1.3.4.i686-linux-musl.tar.gz: d3ba14301b05875188f99872b58fa18e4bde4bac
[ Info: SHA256 of SeqGen.v1.3.4.i686-linux-musl.tar.gz: f8e3c4b89693deb740cd930b68d3b2ba882cebfbd763e661acc4d7050cb63c74
[ Info: Timings: setup: 7.94s, build: 0.7s, audit: 5.36s, packaging: 0.65s
Uploading artifact to https://julia-bb-buildcache.s3.amazonaws.com/69a77d3fd53f882611e745888d144a48e1db5489821a5d02ad708fd7a7aec599/3a1ef2f08b92aa7d2fed22c5a800b3cd80a6955926130bdc2dfde73517ce44b0/i686-linux-musl.tar.gz
curl: Can't open './products/seqgen*i686-linux-musl*.tar.gz'!
curl: try 'curl --help' or 'curl --manual' for more information
##[error]Bash exited with code '26'.
Finishing: run build_tarballs.jl
```

curl can't seem to find the tarball, am I doing something stupid? 

This is the first recipe I have attempted to add.

File: S/seqgen/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+
+name = "SeqGen"
```
Comment (giordano) [inline S/seqgen/build_tarballs.jl:5]: This name must match _exactly_ the directory name, including the case. I'd suggest renaming the directory.

File: S/seqgen/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+# move to bindir
+mkdir ${bindir}
+
+if [[ "${target}" == *-mingw32* ]]; then
+    # not entirely sure this will work on windows
+    # but it is wourth a shot
+    cp seq-gen ${bindir}/seq-gen.exe
+else
+    cp seq-gen ${bindir}/seq-gen
+fi
```
Comment (giordano) [inline S/seqgen/build_tarballs.jl:31]: You can simplify this a lot by using the [automatic environment variables](https://docs.binarybuilder.org/stable/build_tips/#Automatic-environment-variables)
```suggestion
install -Dvm 755 "seq-gen${exeext}" "${bindir}/seq-gen${exeext}"
```

File: S/seqgen/build_tarballs.jl (hunk 1 above)
Comment (BenjaminDoran) [reply S/seqgen/build_tarballs.jl:5]: Thanks for the tips!

File: S/SeqGen/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+# platforms are passed in on the command line
+platforms = supported_platforms()
+
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("seq-gen", :seqgen),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
```
Comment (giordano) [inline S/SeqGen/build_tarballs.jl:36]:
```suggestion
dependencies = Dependency[
```

File: S/SeqGen/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6")
```
Comment (giordano) [inline S/SeqGen/build_tarballs.jl:40]:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6")

```

**Pull request merged**